main/src/cgeo/geocaching/storage/DataStore.java

```diff
@@ -229,6 +237,12 @@ public String call(final Cursor cursor) {
             + "title text not null, "
             + "updated long not null"
             + "); ";
+    private static final String dbCreateCachesLists = ""
+            + "create table " + dbTableCachesLists + " ("
+            + "list_id integer not null, "
+            + "geocode text not null, "
+            + "primary key (list_id, geocode)"
```

Comment:
Search by list_id should be handled by the combined primary key. But to assist the search via geocode an index would be good on the geocode column, yes.